client/app/telecom/telephony/alias/configuration/feature/redirect/telecom-telephony-alias-configuration-redirect.controller.js

```diff
@@ -55,15 +57,27 @@ angular.module('managerApp').controller('TelecomTelephonyAliasConfigurationRedir
     return this.tucVoipServiceAlias.changeDestinationRedirectNumber({
       billingAccount: this.billingAccount,
       serviceName: this.serviceName,
-    }, serviceName);
+    }, serviceName, this.featureTypeToUse);
   }
 
   setNewDestination() {
     return (service) => {
+      this.loading = true;
       this.newDestination = service;
       if (this.canChangeDestination()) {
         this.destinationUsedAsPresentation = false;
       }
+
+      return this.tucVoipServiceLine.getPhone({
+        billingAccount: service.billingAccount,
+        serviceName: service.serviceName,
+      }).then(() => {
+        this.featureTypeToUse = 'redirect';
+      }).catch(() => {
+        this.featureTypeToUse = 'ddi';
```

Comment:
I suppose this works if it's a `404` but what if it's a `504` for instance ?

Reply:
Good point :)